core/src/main/java/org/elasticsearch/rest/action/admin/indices/warmer/put/RestPutWarmerAction.java

```diff
@@ -58,12 +71,17 @@ public RestPutWarmerAction(Settings settings, RestController controller, Client
     }
 
     @Override
-    public void handleRequest(final RestRequest request, final RestChannel channel, final Client client) {
+    public void handleRequest(final RestRequest request, final RestChannel channel, final Client client) throws IOException {
         PutWarmerRequest putWarmerRequest = new PutWarmerRequest(request.param("name"));
+
+        BytesReference sourceBytes = RestActions.getRestContent(request);
+        XContentParser parser = XContentFactory.xContent(sourceBytes).createParser(sourceBytes);
+        QueryParseContext queryParseContext = new QueryParseContext(queryRegistry);
+        queryParseContext.reset(parser);
+        SearchSourceBuilder source = SearchSourceBuilder.PROTOTYPE.fromXContent(parser, queryParseContext);
```

Comment:
Maybe instead we should have a static method in RestActions called getRestSearchSource?